### ISSUE TITLE
Disambiguate offset_polygon_2() from create_offset_polygons_2() 

### ISSUE THREAD

**Issue by afabri** [CGAL/cgal#371]
The first function is in 2D Minkowski Sums, the second function in Straight Skeletons.

We suggest to replace the latter by straight_skeleton_offset_polygons_2() 

We stick to plural as it does not generate a single polygon with holes.
We drop the prefix create_  as that is what functions do.

**Comment by sgiraudot** [comment]
@afabri Can you take care of this?

**Comment by efifogel** [comment]
Look at the entire set of functions that this package provide.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by sgiraudot** [comment]
Good point, it seems that both packages have their own name standards and we can't change one single function without changing the entire sets of function of one or the other packages (which seems an overkill to me).

Unless we can figure out a way of disambiguating these two functions without breaking the consistency of packages, I will close this issue.

**Comment by sgiraudot** [comment]
As no one suggested a better solution, I am closing this issue. Please open a small feature if you want to change the API.

**Comment by HoffmannMichael** [comment]
It is still an issue and it is unresolved. So I don’t see what is gained by closing it.

On 28 Oct 2016, at 07:58, Simon Giraudot <notifications@github.com<mailto:notifications@github.com>> wrote:


As no one suggested a better solution, I am closing this issue. Please open a small feature if you want to change the API.

**Comment by sgiraudot** [comment]
Alright, I'm reopening the issue but I don't see what is gained either by leaving an issue opened and not doing anything to solve it: this issue has been opened for a year with barely anyone participating. So far, the only proposition was from @afabri:

> We suggest to replace the latter by straight_skeleton_offset_polygons_2() 

Which is, as @efifogel pointed, not a valid solution as it breaks the consistency within the [Straight Skeleton package](http://doc.cgal.org/latest/Straight_skeleton_2/group__PkgStraightSkeleton2Functions.html#gaf85a09dc47cb9bb9e2ef28e00fc905ff). In my opinion, the fact that this is an issue is debatable, both packages have their own name standards and it seems unlikely that a user will mistake one from the other, but I'm not in the best position to judge.

@efifogel and @fcacciola, you are the respective maintainers of `Straight_skeleton_2` and of `Minkowski_sum_2`: do you agree there is problem and if so, do you see any way we could solve it?

**Comment by efifogel** [comment]
There is a problem.
BW, it's not only with offset_polygon_2(): A function called
intersection_2() exists in the Kernel and in the 2D Boolean Op. package.
There are 2 possible solutions I can think of: renaming and adding
namespaces (or both).
Introducing namespaces can be done globally to all package in a new major
version.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by HoffmannMichael** [comment]
> On 29 Oct 2016, at 09:17, Efi Fogel <notifications@github.com> wrote:
> 
> There is a problem.
> BW, it's not only with offset_polygon_2(): A function called
> intersection_2() exists in the Kernel and in the 2D Boolean Op. package.

This is less problematic because the intersection operation is conceptually the same in both cases. Otoh, the two offset_polygons are quite different...

When we discussed the issue first, we thought that renaming the straight skeleton incarnation makes sense because this *is* a straight skeleton offset (i.e. very specific to the straight skeleton) rather than a polygon (which, of course, it also happens to be as well, somehow).

**Comment by efifogel** [comment]
On Sat, Oct 29, 2016 at 10:42 AM, Michael Hoffmann <notifications@github.com
> wrote:

>
> > On 29 Oct 2016, at 09:17, Efi Fogel <notifications@github.com> wrote:
> >
> > There is a problem.
> > BW, it's not only with offset_polygon_2(): A function called
> > intersection_2() exists in the Kernel and in the 2D Boolean Op. package.
>
> This is less problematic because the intersection operation is
> conceptually the same in both cases.


Actually, they are not; the version in the "2D Regularized Boolean
Operations" is regularized (closure(interior(<the real intersection>)).

**Comment by HoffmannMichael** [comment]
> > > There is a problem.
> > > BW, it's not only with offset_polygon_2(): A function called
> > > intersection_2() exists in the Kernel and in the 2D Boolean Op. package.
> >
> > This is less problematic because the intersection operation is
> > conceptually the same in both cases.
> 
> 
> Actually, they are not; the version in the "2D Regularized Boolean
> Operations" is regularized (closure(interior(<the real intersection>)).

then maybe it should be renamed to regularized_intersection ?...

**Comment by efifogel** [comment]
On Sat, Oct 29, 2016 at 11:35 AM, Michael Hoffmann <notifications@github.com
> wrote:

> > > > There is a problem.
> > > > BW, it's not only with offset_polygon_2(): A function called
> > > > intersection_2() exists in the Kernel and in the 2D Boolean Op.
> package.
> > >
> > > This is less problematic because the intersection operation is
> > > conceptually the same in both cases.
> >
> >
> > Actually, they are not; the version in the "2D Regularized Boolean
> > Operations" is regularized (closure(interior(<the real intersection>)).
>
> then maybe it should be renamed to regularized_intersection ?...


Yes, perhaps.
All functions there, and there are many, perform regularized operations.
Non have the "regularized_" prefix.
I think it would be better introducing a namespace. (Then, naturally, we
need to keep versions without athe namespace for some time for backward
compatibility.)

**Comment by fcacciola** [comment]
I almost missed this,


On Sat, Oct 29, 2016 at 6:33 AM, Efi Fogel <notifications@github.com> wrote:

> On Sat, Oct 29, 2016 at 11:35 AM, Michael Hoffmann <
> notifications@github.com
> > wrote:
>
> > > > > There is a problem.
> > > > > BW, it's not only with offset_polygon_2(): A function called
> > > > > intersection_2() exists in the Kernel and in the 2D Boolean Op.
> > package.
> > > >
> > > > This is less problematic because the intersection operation is
> > > > conceptually the same in both cases.
> > >
> > >
> > > Actually, they are not; the version in the "2D Regularized Boolean
> > > Operations" is regularized (closure(interior(<the real intersection>)).
> >
> > then maybe it should be renamed to regularized_intersection ?...
>
>
> Yes, perhaps.
> All functions there, and there are many, perform regularized operations.
> Non have the "regularized_" prefix.
> I think it would be better introducing a namespace. (Then, naturally, we
> need to keep versions without athe namespace for some time for backward
> compatibility.)
>
>
> I agree this really is a problem, and I think introducing a namespace
would be the a reasonable solution.

But I'm not sure how to effectively go about this, in terms of backward
compatibity.

Best


-- 
Fernando Cacciola
SciSoft Consulting, Founder
http://www.scisoft-consulting.com

**Comment by lrineau** [comment]
If a namespace is introduced and we keep the names of functions and classes, using-declarations can be used to keep the backward-compatibility.

**Comment by sloriot** [comment]
@efifogel Shall we introduce the new namespace for BO_2 functions while breaking compatibility in CGAL 5.0?
